packages/language-server/src/server.ts

```diff
@@ -361,16 +346,7 @@ documents.onDidChangeContent(async (changeEvent): Promise<void> => {
       isFallback,
       isIncorrectVersion,
     });
-    validateOpenDocuments(documentUri);
-    const diagnostics = getXMLViewDiagnostics({
-      document,
-      context,
-    });
-    getLogger().trace("computed diagnostics", { diagnostics });
-    connection.sendDiagnostics({
-      uri: changeEvent.document.uri,
-      diagnostics,
-    });
+    await validateOpenDocuments();
```

Comment:
Do we really need to revalidate all documents on each change? We have different type of validators and currently we only have one that needs to be run for every change - unique ids. But we do not need to run other validators for every single open file.

Reply:
yes it is needed. The scenario changed with validation of unique ID across xml view files. Now we need to validate other opened documents as uniqueness of ID depends on other file. Now when user change content of xml view file, we need to collect diagnostics for all open documents and send these diagnostics to client for each xml document view. Here all validation should be involved as user may have other diagnostics apart from ID.  Short video illustrate basic follow
![2024-08-21_08-27-44 (1)](https://github.com/user-attachments/assets/45f3d152-d67e-43d0-a151-940a3ce03c80)

Here we change `id` of `Text` in `App.view.xml` file. This change affect `InvoiceList.view.xml` file and vise versa. Furthermore validation of `Button` e.g.`The "Button" class can't have an empty ID attribute when flexEnabled is "true" in manifest.json.` need to be collected which involves other validations.